### DISCUSSION TITLE
Guarantee of delivery

### DISCUSSION THREAD

**Discussion by sushiMix** [open-telemetry/opentelemetry-collector#8271] [Q&A] [answered]
Hello, I 'm trying to determine If can can not lose any message in the telemetry collector (i'm looking for logs which is still in developpement).
I did a simple case with a receiver, a custom processor (which adds an attribute) and an exporter (which can not call the target server).
The receiver returns me a 200Ok whereas i'm sure it can't forward log. 

Is there a way to configure a guarantee of delivery (I assume the batch processor is not ready to persist and restart with inflight data) when a direct "receiver/exporter" configuration is performed (is it to the receiver/exporter to implement such feature) ?

**Comment by andrzej-stencel** [comment, marked as answer]
If you need a guarantee of delivery, you should probably not use the Batch processor in its current form. Here's a "parent" issue that describes the problem with the Batch processor and a couple other problems in related issues:
- https://github.com/open-telemetry/opentelemetry-collector/issues/7460

**Reply by sushiMix** [reply]
Hello, thanks for the parent issue, I encountered the other problems as tried without batch processor.
